docling/types.py

```diff
@@ -0,0 +1,8 @@
+"""Shared type definitions for the docling package."""
+
+from typing import Any, Dict, Type, Union
+
+from pydantic import BaseModel
+
+# Type alias for template parameters that can be string, dict, or BaseModel
+ExtractionTemplateType = Union[str, Dict[str, Any], BaseModel, Type[BaseModel]]
```

Comment:
don't we have the `docling.datamodel` module for all of this?

Reply:
I suggest moving this and `ExtractionResult` + `ExtractedPageData` to `docling.datamodel.extraction`.

Since we separate DocumentConvert from DocumentExtractor it makes sense the result objects are in separate modules.